ruby/example_code/cloudtrail/aws-ruby-sdk-cloudtrail-example-lookup-events.rb

```diff
@@ -21,32 +21,90 @@
 # OF ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-require 'aws-sdk-cloudtrail'  # v2: require 'aws-sdk'
+require 'aws-sdk-cloudtrail'
 
-def show_event(event)
-  puts 'Event name:   ' + event.event_name
-  puts 'Event ID:     ' + event.event_id
-  puts "Event time:   #{event.event_time}"
-  puts 'User name:    ' + event.username
+# Lists selected information about existing events in CloudTrail.
+class ListEventsExample
+  # Initialize an instance of ListEventsExample, creating a client for
+  # AWS CloudTrail (unless already provided during initialization).
+  #
+  # (The following comments express documentation about this function in YARD 
+  # format by using @ symbols.)
+  #
+  # @param [Hash] opts ({}) A hash of an API client for CloudTrail.
+  # @option [Aws::CloudTrail::Client] :cloudtrail_client
+  #  (Aws::CloudTrail::Client)
+  def initialize(opts = {})
+    # This CloudTrail API client is used to list the CloudTrail resources.
+    @cloudtrail = opts[:cloudtrail_client] || Aws::CloudTrail::Client.new
+  end
 
-  puts 'Resources:'
+  # Lists selected information about existing events in AWS CloudTrail.
+  def list_events()
+    resp = @cloudtrail.lookup_events()

```

Comment:
When you call methods without params, there's no need to include `()` in fact it's generally not recommended unless you need to explicitly send no arguments (i.e. [`super` vs `super()`](https://stackoverflow.com/questions/31816149/difference-between-calling-super-and-calling-super))